### PR TITLE
Fixed: reduce query size and remove rdflib dependency

### LINKED ISSUE #16: Queries are too long
I've been reported that the queries done by this service are too long (>10k bytes) and apparently such queries fail on Virtuoso.

The culprit apparently is this one:

```
PREFIX ...
WITH <http://mu.semte.ch/application>
        INSERT DATA {
<dockevent:501c772bb31e062499be22c2d7e6617fd3e67c0bcc3afd491edcb4e38a4c4e18_1500559097170790847> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://ontology.aksw.org/dockevent/types/event> .
<dockevent:501c772bb31e062499be22c2d7e6617fd3e67c0bcc3afd491edcb4e38a4c4e18_1500559097170790847> <http://ontology.aksw.org/dockevent/eventId> "501c772bb31e062499be22c2d7e6617fd3e67c0bcc3afd491edcb4e38a4c4e18_1500559097170790847" .
<dockevent:501c772bb31e062499be22c2d7e6617fd3e67c0bcc3afd491edcb4e38a4c4e18_1500559097170790847> <http://ontology.aksw.org/dockevent/type> <http://ontology.aksw.org/dockevent/types/container> .
...

<dockcontainer_network:4733fc3af0f1fad45413685f35b627cd55bbcb5bbfcaa9fd3646b8ecd179885b_1500559097170790847> <http://ontology.aksw.org/dockevent/link> "appbdepipeline_db_1:appbdepipeline_db_1" .
<dockcontainer:501c772bb31e062499be22c2d7e6617fd3e67c0bcc3afd491edcb4e38a4c4e18_1500559097170790847> <http://ontology.aksw.org/dockcontainer/env> "MAIN_APP_FILE=web.rb" .
...

<dockcontainer:501c772bb31e062499be22c2d7e6617fd3e67c0bcc3afd491edcb4e38a4c4e18_1500559097170790847> <http://ontology.aksw.org/dockcontainer/label> "com.docker.compose.oneoff=False" .
<dockcontainer:501c772bb31e062499be22c2d7e6617fd3e67c0bcc3afd491edcb4e38a4c4e18_1500559097170790847> <http://ontology.aksw.org/dockcontainer/env> "RUBYGEMS_VERSION=2.6.4" .
...

<dockcontainer:501c772bb31e062499be22c2d7e6617fd3e67c0bcc3afd491edcb4e38a4c4e18_1500559097170790847> <http://ontology.aksw.org/dockcontainer/env> "MU_SPARQL_ENDPOINT=http://database:8890/sparql" .
<dockcontainer:501c772bb31e062499be22c2d7e6617fd3e67c0bcc3afd491edcb4e38a4c4e18_1500559097170790847> <http://ontology.aksw.org/dockcontainer/label> "com.docker.compose.version=1.13.0" .
...

<dockevent_actors:501c772bb31e062499be22c2d7e6617fd3e67c0bcc3afd491edcb4e38a4c4e18_1500559097170790847> <http://ontology.aksw.org/dockevent/nodeIpPort> "" .
<dockevent_actors:501c772bb31e062499be22c2d7e6617fd3e67c0bcc3afd491edcb4e38a4c4e18_1500559097170790847> <http://ontology.aksw.org/dockevent/nodeName> "" .
...
}
```

Which I believe come from the output of rdflib. I will make a PR to convert the code to aiosparql. This will make a better use of the `;` syntax and I can probably improve the use of prefixes.

### PR DESCRIPTION
**Pull request by cecton**
Fixes #16

### CODE REVIEW

File: muswarmlogger/event2rdf.py [hunk 1]
```diff
@@ -36,99 +20,79 @@ def add_event_to_graph(self, event, container=None):
...
-                container_node.add(DOCKCONTAINER.network, network_node)
+                        network_node.append((DockEvent.link, link))
+                container_node.append((DockContainer.network, network_node))
 
         actor = event.get("Actor", "")
         if actor != "":
             actor_id = actor.get("ID", "")
             actor_id = "%s_%s" % (actor_id, _timeNano)
-            actor_node = self.store.resource("dockevent_actors:%s" % actor_id)
-            actor_node.add(DOCKEVENT.actorId, Literal(actor_id, datatype=XSD.dateTime))
```
Comment (cecton) [inline muswarmlogger/event2rdf.py:112]: I removed the datatype here because it was entirely wrong.

File: muswarmlogger/event2rdf.py [hunk 2]
```diff
@@ -36,99 +20,79 @@ def add_event_to_graph(self, event, container=None):
...
-        if event_type == "container":
-            event_node.add(DOCKEVENT.type, DOCKEVENT_TYPES.container)
-        elif event_type == "network":
-            event_node.add(DOCKEVENT.type, DOCKEVENT_TYPES.network)
-        elif event_type == "plugin":
-            event_node.add(DOCKEVENT.type, DOCKEVENT_TYPES.plugin)
-        elif event_type == "volume":
-            event_node.add(DOCKEVENT.type, DOCKEVENT_TYPES.volume)
+        event_node.append(
+            (DockEvent.type, getattr(DockEventTypes, event_type)))
```
Comment (cecton) [inline muswarmlogger/event2rdf.py:33]: Compared to the old code, it will break if the type is not implemented.

File: muswarmlogger/event2rdf.py [hunk 3]
```diff
@@ -36,99 +20,79 @@ def add_event_to_graph(self, event, container=None):
...
-        elif event_action_type == "exec_start":
-            event_node.add(DOCKEVENT.action, DOCKEVENT_ACTIONS.exec_start)
-        elif event_action_type == "health_status":
-            event_node.add(DOCKEVENT.action, DOCKEVENT_ACTIONS.health_status)
-        elif event_action_type == "die":
-            event_node.add(DOCKEVENT.action, DOCKEVENT_ACTIONS.die)
-        elif event_action_type == "destroy":
-            event_node.add(DOCKEVENT.action, DOCKEVENT_ACTIONS.destroy)
+        event_node.append(
+            (DockEvent.action, getattr(DockEventActions, event_action_type)))
```
Comment (cecton) [inline muswarmlogger/event2rdf.py:44]: It will break if the action type is not implemented.

File: muswarmlogger/event2rdf.py [hunk 4]
```diff
@@ -1,30 +1,14 @@
...
+from muswarmlogger.prefixes import (
+    DockContainer, DockContainerNetwork, DockEvent, DockEventActions,
+    DockEventTypes)
 
-DOCKEVENT = Namespace('http://ontology.aksw.org/dockevent/')
-DOCKEVENT_TYPES = Namespace('http://ontology.aksw.org/dockevent/types/')
-DOCKEVENT_ACTIONS = Namespace('http://ontology.aksw.org/dockevent/actions/')
-DOCKEVENT_ACTORS = Namespace('http://ontology.aksw.org/dockevent/actors/')
-DOCKCONTAINER = Namespace('http://ontology.aksw.org/dockcontainer/')
-DOCKCONTAINER_NETWORK = Namespace('http://ontology.aksw.org/dockcontainer/network/')
```
Comment (cecton) [inline muswarmlogger/event2rdf.py:12]: Everything is moved to `prefixes.py`

File: muswarmlogger/loggers/sparql.py [hunk 5]
```diff
@@ -245,16 +245,15 @@
...
-    await sparql.update("""
-        WITH <%(graph)s>
+    triples = e2rdf.serialize()
+    await sparql.update(
+        """
         INSERT DATA {
-            %(ntriples)s
+            GRAPH {{graph}} {
+                {{}}
+            }
```
Comment (cecton) [inline muswarmlogger/loggers/sparql.py:254]: `WITH <...> INSERT DATA` does not exist in standard SPARQL. It's `INSERT DATA { GRAPH <...>`

File: muswarmlogger/loggers/sparql.py [hunk 6]
```diff
@@ -245,16 +245,15 @@
...
         }
-        """ % {
-            "graph": graph,
-            "ntriples": ntriples,
-        })
+        """, triples)
```
Comment (cecton) [inline muswarmlogger/loggers/sparql.py:256]: The graph is now taken from the SPARQLClient. This is the same as the environment variable. I'm not going to cleanup the old way in this PR.

File: muswarmlogger/event2rdf.py [hunk 7]
```diff
@@ -1,30 +1,14 @@
...
-from rdflib.namespace import OWL, XSD, RDFS
-from rdflib.namespace import Namespace
+from muswarmlogger.prefixes import (
+    DockContainer, DockContainerNetwork, DockEvent, DockEventActions,
+    DockEventTypes)
 
-DOCKEVENT = Namespace('http://ontology.aksw.org/dockevent/')
-DOCKEVENT_TYPES = Namespace('http://ontology.aksw.org/dockevent/types/')
-DOCKEVENT_ACTIONS = Namespace('http://ontology.aksw.org/dockevent/actions/')
-DOCKEVENT_ACTORS = Namespace('http://ontology.aksw.org/dockevent/actors/')
```
Comment (cecton) [inline muswarmlogger/event2rdf.py:10]: Wasn't used, I dropped it.

**Comment by cecton** [comment]
I just received the information that the change fixed the problem as expected. I'm waiting on your reviews :)

**Review by earthquakesan — APPROVED** [review]
lgtm

**Review by fr0gs — APPROVED** [review]
It looks ok to me.

**Pull request merged**